editor/editor.gd

```diff
@@ -70,29 +76,44 @@ func _process(_delta: float) -> void:
 			timeline.value = root_viewport.music.get_playback_position() * 100.0 / root_viewport.music.stream.get_length()
 		
 		#timestamp schenanigans
-		var time = root_viewport.music.get_playback_position()
-		var mils = fmod(time,1)*1000
-		var secs = fmod(time,60)
-		var mins = fmod(time, 60*60) / 60
+		current_music_time += elapsed_os_time * current_music_speed
+		var mils = fmod(current_music_time,1)*1000
+		var secs = fmod(current_music_time,60)
+		var mins = fmod(current_music_time, 60*60) / 60
 		timestamp.text = "%02d:%02d:%03d" % [mins,secs,mils]
-		cur_time = time
 	
-	if prev_time != cur_time:
-		for i in range(obj_container.get_child_count() - 1, -1, -1):
-			## Comment
-			var hit_object := obj_container.get_child(i) as HitObject
-
-			hit_object.move(bar_right.rect_size.x, cur_time)
+	if current_music_time != previous_music_time:
+		updateNotes()
 
 	var mouse_pos := get_viewport().get_mouse_position().x
-	var raw_pos := mouse_pos - hit_point.rect_position.x
-	var new_timing := cur_time + (raw_pos / current_velocity / DEFAULT_VELOCITY)
-	var cur_bpm = get_current_bpm(new_timing)
+	var raw_pos := mouse_pos - hit_point.rect_position.x - (hit_point.rect_size.x / 2)
+	
+	var new_timing := 0.0
+	if using_constant_sv:
+		new_timing = current_music_time + (raw_pos / current_velocity / DEFAULT_VELOCITY)
+	else:
+		var tracking_speed := 0.0
+		for i in range(obj_container.get_child_count() - 1, -1, -1):
+			var hit_object := obj_container.get_child(i) as HitObject
+			if hit_object.rect_position.x > raw_pos:
+				break
+			tracking_speed = hit_object.speed
```

Comment:
@FuzzyFus from what we've talked about, we don't actually want a placed `HitObject` to inherit the SV of a previous `HitObject`, but rather inherit a base SV instead. I could be wrong, but likely you won't want your snappings to randomly change without some sort of feedback.

Reply:
its a cool implementation but placing notes in sv mode is super unintuitive, ill probably just disable notes from being placed in sv mode to make it easier on everyone

Reply:
I think at some point I tried filtering out just the Timing Points but for some reason there were only like 3 or 4, when in fact there are like 700 in the map I was testing. I thought our HitObject had an SV value whether it was a timing point or a note? but yeah, it's gonna look weird either way, even with a better implementation